src/cli/src/export.rs

```diff
@@ -375,84 +485,127 @@ impl Export {
                 let _permit = semaphore_moved.acquire().await.unwrap();
-                let db_dir = self.catalog_path().join(format!("{schema}/"));
+
+                let db_dir = export_self.catalog_path().join(&schema);
                 tokio::fs::create_dir_all(&db_dir)
                     .await
                     .context(FileIoSnafu)?;
 
-                let with_options = match (&self.start_time, &self.end_time) {
-                    (Some(start_time), Some(end_time)) => {
-                        format!(
-                            "WITH (FORMAT='parquet', start_time='{}', end_time='{}')",
-                            start_time, end_time
-                        )
-                    }
-                    (Some(start_time), None) => {
-                        format!("WITH (FORMAT='parquet', start_time='{}')", start_time)
-                    }
-                    (None, Some(end_time)) => {
-                        format!("WITH (FORMAT='parquet', end_time='{}')", end_time)
-                    }
-                    (None, None) => "WITH (FORMAT='parquet')".to_string(),
+                let (path, connection_part) = if export_self.s3 {
+                    let s3_path = format!(
+                        "s3://{}/{}/{}/",
+                        export_self.s3_bucket.as_ref().unwrap(),
+                        export_self.catalog,
+                        schema
+                    );
+                    // endpoint is optional
+                    let endpoint_option = if let Some(endpoint) = export_self.s3_endpoint.as_ref() {
+                        format!(", ENDPOINT='{}'", endpoint)
+                    } else {
+                        String::new()
+                    };
+                    let connection_options = format!(
+                        "ACCESS_KEY_ID='{}', SECRET_ACCESS_KEY='{}', REGION='{}'{}",
+                        export_self.s3_access_key.as_ref().unwrap(),
+                        export_self.s3_secret_key.as_ref().unwrap(),
+                        export_self.s3_region.as_ref().unwrap(),
+                        endpoint_option
+                    );
+                    (s3_path, format!(" CONNECTION ({})", connection_options))
+                } else {
+                    (
+                        export_self
+                            .catalog_path()
+                            .join(format!("{schema}/"))
+                            .to_string_lossy()
+                            .to_string(),
+                        String::new(),
+                    )
                 };
 
                 let sql = format!(
-                    r#"COPY DATABASE "{}"."{}" TO '{}' {};"#,
-                    self.catalog,
-                    schema,
-                    db_dir.to_str().unwrap(),
-                    with_options
+                    r#"COPY DATABASE "{}"."{}" TO '{}' WITH ({}){};"#,
+                    export_self.catalog, schema, path, with_options_clone, connection_part
```

Comment:
@fengjiachun do we support copy database with connection to all `OSS`?
I checked the doc only find s3
will check the code later

Reply:
Yes, the `copy database` supports connecting to both `oss` and `s3` using the same arguments: https://docs.greptime.com/reference/sql/copy/#options

Reply:
just check the code about copy parquet part for now its only support s3 and f3
https://github.com/GreptimeTeam/greptimedb/blob/9860bca986c3639468d261468802930e8dec79d2/src/common/datasource/src/object_store.rs#L56-L75

So I think just support s3 is OK and maybe support others in the future

Reply:
Oh my fault, let's just consider s3 for now.

Reply:
forget to support two things the latest commit support them, I think this is ready for review